### PR TITLE
BLD : update in-repo recipe

### PR DESCRIPTION
**Pull request by tacaswell**
- Updated the version string to consult the git ENVs
 - install everything from bin into the prefix

### CODE REVIEW

File: conda_build.recipe/meta.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ build:
 requirements:
   build:
     - python
+    - conda
```
Comment (asmeurer) [inline conda_build.recipe/meta.yaml]: You can't install conda into a build environment.

Comment (asmeurer) [inline conda_build.recipe/meta.yaml]: In fact, I'm not sure how this recipe works. @tswicegood

File: conda_build.recipe/build.sh [hunk 2]
```diff
@@ -3,3 +3,4 @@
 $PYTHON setup.py install
 
 cp bdist_conda.py ${PREFIX}/lib/python${PY_VER}/distutils/command
+install bin/* ${PREFIX}/bin
```
Comment (asmeurer) [inline conda_build.recipe/build.sh]: Doesn't setup.py already do this?

Comment (tacaswell) [inline conda_build.recipe/build.sh]: If I didn't add this line the `conda-build -h` test was failing with

```
===== testing package: conda-build-1.12.0.post2-py27_2_gd3653ce =====
import: u'conda_build'
running run_test.py
conda_build.__version__: unknown
===== conda-build-1.12.0.post2-py27_2_gd3653ce OK =====
+ conda-build -h
Traceback (most recent call last):
  File "/home/tcaswell/.pe/_test/bin/conda-build", line 4, in <module>
    __import__('pkg_resources').run_script('conda-build===1.12.0-2-gd3653ce', 'conda-build')
  File "/home/tcaswell/.pe/_test/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/pkg_resources/__init__.py", line 723, in run_script
  File "/home/tcaswell/.pe/_test/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/pkg_resources/__init__.py", line 1628, in run_script
pkg_resources.ResolutionError: No script named 'conda-build'
TESTS FAILED: conda-build-1.12.0.post2-py27_2_gd3653ce
```

File: conda_build.recipe/meta.yaml (hunk 1 above)
Comment (tacaswell) [inline conda_build.recipe/meta.yaml]: With out this dep you get

```

Installed /home/tcaswell/.pe/_build/lib/python2.7/site-packages/conda_build-1.12.0_7_g8468d76-py2.7.egg
Processing dependencies for conda-build===1.12.0-7-g8468d76
Searching for conda
Traceback (most recent call last):
  File "setup.py", line 54, in <module>
    package_data={'conda_build': ['templates/*', 'cli-*.exe']},
  File "/home/tcaswell/.pe/_build/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/home/tcaswell/.pe/_build/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/home/tcaswell/.pe/_build/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/install.py", line 67, in run
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/install.py", line 117, in do_egg_install
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/easy_install.py", line 380, in run
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/easy_install.py", line 610, in easy_install
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/easy_install.py", line 661, in install_item
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/easy_install.py", line 709, in process_distribution
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/pkg_resources/__init__.py", line 824, in resolve
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/pkg_resources/__init__.py", line 1069, in best_match
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/pkg_resources/__init__.py", line 1081, in obtain
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/command/easy_install.py", line 617, in easy_install
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/package_index.py", line 594, in fetch_distribution
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/package_index.py", line 431, in find_packages
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/package_index.py", line 765, in scan_url
  File "/home/tcaswell/.pe/_build/lib/python2.7/site-packages/setuptools-15.0-py2.7.egg/setuptools/package_index.py", line 283, in process_url
RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.python.org/simple/conda/r
Command failed: /bin/bash -x -e /home/tcaswell/other_source/conda-build/conda_build.recipe/build.sh
```

It looks like the setup.py requires it so it is trying to reach out to pip to install it.

File: conda_build.recipe/build.sh (hunk 2 above)
Comment (tacaswell) [inline conda_build.recipe/build.sh]: And yes, it does look like setup.py does do this (and spams to the screen that it is installing).  I wonder if the issue is a missing `hash -r` call as the testing env does not get properly activated.  Does `install` do a bit more magic under the hood?

Comment (asmeurer) [inline conda_build.recipe/build.sh]: I suspect setuptools is to blame here.

Comment (asmeurer) [inline conda_build.recipe/build.sh]: Does 

```diff
diff --git a/setup.py b/setup.py
index 0503427..cc1ac88 100755
--- a/setup.py
+++ b/setup.py
@@ -1,9 +1,9 @@
 #!/usr/bin/env python
 import sys

-try:
+if 'develop' in sys.argv:
     from setuptools import setup
-except ImportError:
+else:
     from distutils.core import setup

 import versioneer
```

make any difference?

**Comment by tacaswell** [comment]
Yes, that seems to have fixed it.

File: conda_build.recipe/meta.yaml (hunk 1 above)
Comment (tswicegood) [inline conda_build.recipe/meta.yaml]: @asmeurer All the `build.sh` file did was call `setup.py` and copy the `bdist_conda` file into the correct place.  In a closer to ideal world (i.e., this world: https://github.com/conda/conda/issues/1145) this package does have `conda` as one of the packages it depends, but that can't be done right now so I didn't include it.

So far as I know, this line isn't needed.

**Comment by msarahan** [comment]
superseded by later work and #924.  Thanks @tacaswell for putting this forward.

**Pull request closed without merging**